tasks/nunjucks.js

```diff
@@ -6,65 +6,159 @@
+                    return reject('For some destinations were not specified source files.')
                 }
-                grunt.file.write(f.dest, res);
-                grunt.log.writeln('File "' + f.dest + '" created.');
-                done();
-            });
-
-        }, function(err) {
-            if (err) {
-                grunt.log.error(err);
-                grunt.fail.warn('Something went wrong.');
+
+                // Iterate over files' sources
+                file.src.forEach(src => {
+                    // Сheck whether source file exists
+                    if (!grunt.file.exists(src)) {
+                        grunt.log.error(`Source file ${chalk.cyan(src)} for ${chalk.cyan(filedest)} not found.`)
+
+                        // Skip to next source file — nothing we can do with non-existing file
+                        return reject('Some source files were not found.')
+                    }
+
+                    // Construct absolute path to file for Nunjucks
+                    let filepath = path.join(process.cwd(), src)
+
+                    let data = {}
+                    // Work with data only if there is any data
+                    if (options.data) {
+                        // Clone data
+                        for (let i in options.data) {
+                            if (options.data.hasOwnProperty(i)) {
+                                data[i] = options.data[i]
+                            }
+                        }
+
+                        // Preprocess data
+                        if (typeof options.preprocessData === 'function') {
+                            data = options.preprocessData.call(file, data)
+                        }
+                    }
+
+                    // Asynchronously render templates with configurated Nunjucks environment
+                    // and write to destination
+                    env.render(filepath, data, (error, result) => {
+                        // Catch errors, warn
+                        if (error) {
+                            grunt.log.error(error)
+                            grunt.fail.warn('Failed to compile one of the source files.')
+                            grunt.log.writeln()
+
+                            // Prevent writing of failed to compile file, skip to next file
+                            return reject('Failed to compile some source files.')
+                        }
+
+                        // Write rendered template to destination
+                        grunt.file.write(filedest, result)
+
+                        // Debug process
+                        grunt.verbose.ok(`File ${chalk.cyan(filedest)} created.`)
+                        grunt.verbose.writeln()
+
+                        countCompiled++
+                    })
+                })
+            })
+
+            // Finish Promise
+            resolve()
+        })
+
+        // Print any errors from rejects
+        .catch(error => {
```

Comment:
Maybe we can create several promises — one for each file and use [`Promise.all`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) to process them all: `catch` for that promise (returned by `Promise.all`) receives messages from all included promises